nimare/tests/test_estimator_performance.py

```diff
@@ -0,0 +1,268 @@
+        logp_values_data = cres.get_map(
+            "logp_level-voxel_corr-FWE_method-montecarlo", return_type="array"
+        )
+        # transform logp values back into regular p values
+        p_values_data = 10 ** -logp_values_data
+
+        if isinstance(meta, mkda.KDA) and isinstance(kern, kernel.MKDAKernel):
+            # KDA with the MKDAKernel gives p-values twice as small as allowed
+            assert p_values_data.min() >= (1.0 / (corr.parameters["n_iters"] * 2))
+        else:
+            # there should not be p-values less than 1 / n_iters
+            assert p_values_data.min() >= (1.0 / corr.parameters["n_iters"])
+
+        # max p-values should be less than 1
+        assert p_values_data.max() < 1.0
+    else:
+        p_values_data = cres.get_map("p", return_type="array")
+
+    # ensure all values are between 0 and 1 (inclusively)
+    assert (p_values_data >= 0).all() and (p_values_data <= 1).all()
+
+    ################################################################
+    # CHECK IF META/KERNEL/CORRECTOR COMBINATION PERFORMS ADEQUATELY
+    ################################################################
+    # The below combinations cannot even detect
+    # significance at the ground truth foci voxel
+    # locations
+    if (
+        (
+            isinstance(meta, mkda.MKDADensity)
+            and isinstance(kern, kernel.KDAKernel)
+            and corr.method != "montecarlo"
+        )
+        or (
+            isinstance(meta, mkda.MKDADensity)
+            and isinstance(kern, kernel.MKDAKernel)
+            and corr.method != "montecarlo"
+        )
+        or (
+            isinstance(meta, ale.ALE)
+            and isinstance(kern, kernel.MKDAKernel)
+            and corr.method == "montecarlo"
+        )
+    ):
+        good_performance = False
+    else:
+        good_performance = True
+    mask = cres.masker.mask_img
+    ground_truth_foci_ijks = [tuple(mm2vox(focus, mask.affine)) for focus in ground_truth_foci]
+    # reformat coordinate indices to index p_values_data
+    gtf_idx = [
+        [ground_truth_foci_ijks[i][j] for i in range(len(ground_truth_foci_ijks))]
+        for j in range(3)
+    ]
+
+    p_values_volume = cres.masker.inverse_transform(p_values_data).get_fdata()
+    # the ground truth foci should be significant at minimum
+    best_chance_p_values = p_values_volume[gtf_idx]
+    assert all(best_chance_p_values < ALPHA) == good_performance
+
+    # double the radius of what is expected to be significant
+    r = fwhm
+    # create an expectation of significant/non-significant regions
+    sig_map, nonsig_map = _create_signal_mask(mask, ground_truth_foci_ijks, r=r)
+    sig_idx = cres.masker.transform(sig_map).astype(bool).squeeze()
+    nonsig_idx = cres.masker.transform(nonsig_map).astype(bool).squeeze()
+    # while this combination passes the basic
+    # test of significance at the ground truth foci voxels,
+    # it does not meet the criteria for 50% power
```

Comment:
We should probably set two thresholds then: a narrower radius where we're confident we can detect effects, and a looser one outside of which we're confident we won't detect effects (we can ignore the middle zone).

(Alternatively, we can boost the signal or reduce the variability until this is a non-issue.)

Reply:
done! 2 mm threshold for signal and 14 mm threshold for non-signal (noise)